Formbuilder: support rows and columns values for Note (textarea) custom field type

composerjk opened the pull request:
Overview
----------------------------------------
Before Form Builder, defining a Note with rows and columns (defaulting to 4 and 60) would show a `textarea` tag with `rows` and `cols` attributes set. Form Builder does not currently set those attributes.

Briefly mentioned on [Mattermost](https://chat.civicrm.org/civicrm/pl/3kqcc83ymfryjkq6nnqkq84c5a).

Before
----------------------------------------
Displaying a custom field of type `Note` in Form Builder would show the browser default for a `textarea` tag.

After
----------------------------------------
Displaying a custom field of type `Note` in Form Builder now shows  the `textarea` tag with `rows` and `cols` set to the custom field definition set by the admin for that field.

Technical Details
----------------------------------------
In the `civicrm_custom_field` table, there are field names of `note_rows` and `note_columns`. Form Builder was setting `input_attrs` for `rows` but not `columns` since it was looking for `note_cols`. This PR corrects that lookup.

Then, the angular field definition for TextArea adds the `rows` and `cols` attributes reference the settings of `input_attrs.rows` and `input_attrs.columns` with defaults of 4 and 60, respectively, if those attributes were somehow not defined.

Comments
----------------------------------------
Someone should decide if `cols` should also become `columns` to match `note_columns` in the [`foreach` line of SpecFormatter.php](https://github.com/asmac-org/civicrm-core/blob/0bd36ff3dda6c64a16d92236f524b931bab32bb8/Civi/Api4/Service/Spec/SpecFormatter.php#L323).

composerjk committed:
reference correct note_columns field name in database

- civicrm_custom_field table has `note_rows` and `note_columns`, *not* `note_cols`
- Q: Should `cols` also be changed to `columns` on this line?

composerjk committed:
Form Builder: TextArea: use existing rows/cols of custom field definition

eileenmcnaughton commented:
Lets run those tests again to get  a green tick - test this please

colemanw commented:
Thank you @composerjk this is definitely a needed fix.
I would suggest one alteration: since the APIv4 specformatter has the goal of bridging differences between core and custom fields, and it's already looking for a "cols" property of core fields, we don't want the custom fields to come out differently so let's replace `note_columns` with `cols` in the string replace. Then in Afform we can just check for "cols" and not need to handle both.

composerjk committed:
use 'cols' as key instead of 'columns' for note_columns per coleman

As suggested in this comment from @colemanw:
https://github.com/civicrm/civicrm-core/pull/28388#issuecomment-1835717428

Note: This will also turn a prop of `columns` into `cols` for the key not just for a `note_columns` prop.

composerjk commented:
Change made and pushed, @colemanw. Was at a type design conference all day, so couldn't get to it until just now.

Note that with the current implementation, the code will also take a property of `columns` and turn it into a inputAttrs key of `cols` (in addition to `note_columns`).

composerjk committed:
fix for CiviCRM-Core-Style check
